elastic/apm-server
# Check APM Server version in ingest pipeline

## Motivation/summary

We now check `observer.version` represents an APM Server version
no greater than the installed integration package, disregarding
the patch and tag components. This is done in the ingest pipeline,
which is tied to the integration package version.
`observer.version_major` is now set in the ingest pipeline, rather
than in apm-server code.

We introduce an APM-specific application logs data stream. We had
been relying on the base "logs-*" data stream installed by
Elasticsearch. By introducing an APM-specific data stream, we can
define an ILM policy and ingest pipeline which performs the version
check.

## Checklist

~- [ ] Update [CHANGELOG.asciidoc](https://github.com/elastic/apm-server/blob/master/CHANGELOG.asciidoc)~
~- [ ] Documentation has been updated~

## How to test these changes

1. Install integration 8.0.0 or 8.1.0
2. Run a custom apm-server with the version bumped to 9.0.0
3. Send some events to apm-server
4. Check the events are never indexed, no data streams are created
5. Check logs for errors suggesting that the integration should be upgraded

## Related issues

Closes https://github.com/elastic/apm-server/issues/6425
